vocabularies/UI.xml

```diff
@@ -1837,6 +1837,31 @@ This switches off the validation of user input with respect to decimals.</String
         <Annotation Term="Core.Description" String="The referenced entity set is the preferred starting point for UIs using this service" />
       </Term>
 
+      <Term Name="RecursiveHierarchy" BaseTerm="Hierarchy.RecursiveHierarchy" Type="UI.RecursiveHierarchyType" AppliesTo="EntityType">
+        <Annotation Term="Common.Experimental" />
+        <Annotation Term="Core.Description" String="Rules for hierarchy nodes that a UI can guide the user towards" />
+        <Annotation Term="Core.LongDescription">
+          <String>For example, dropping a node on another node can be prevented if the dropped node cannot
+          be a child of the target node.</String>
+        </Annotation>
+      </Term>
+      <ComplexType Name="RecursiveHierarchyType">
+        <Annotation Term="Common.Experimental" />
+        <Property Name="ChildRules" Type="Collection(UI.RecursiveHierarchyPasteType)" Nullable="false">
+          <Annotation Term="Core.Description" String="Which nodes can be children of this node" />
+        </Property>
+      </ComplexType>
+      <ComplexType Name="RecursiveHierarchyChildType">
+        <Annotation Term="Common.Experimental" />
+        <Annotation Term="Core.Description" String="Rule for nodes of a given type (specified by the [`Hierarchy.RecursiveHierarchy/NodeTypeProperty`](Hierarchy.md#RecursiveHierarchyType))" />
```

Comment:
I didn't find a RecursiveHierarchy/NodeTypeProperty (neither in the SAP Hierarchy vocabulary nor in the OASIS TCS Aggregation vocabulary. Is this something to be defined then?

Reply:
It is in the release candidate branch:
https://github.com/SAP/odata-vocabularies/blob/9500b9cc688cae3eb5ca1b71ac8c9b0d83f1d407/vocabularies/Hierarchy.xml#L89